net/portmapper/upnp.go

```diff
@@ -86,6 +86,7 @@ const tsPortMappingDesc = "tailscale-portmap"
 // addAnyPortMapping abstracts over different UPnP client connections, calling the available
 // AddAnyPortMapping call if available for WAN IP connection v2, otherwise defaulting to the old
 // behavior of calling AddPortMapping with port = 0 to specify a wildcard port.
+// It returns the new external port, or an error.
```

Comment:
why does it just return something the caller already passed in?

why not just return an error?

Reply:
Wait, the caller passes `prevPort` for this argument.

How does this even work? `prevPort` will usually be 0 on startup.

Reply:
It can either return something the caller passed in, or can return an entirely new port from the AddAnyPortMapping. I've updated the docs for that